src/compose.ts

```diff
@@ -274,17 +281,99 @@ function normalizeArrayOfStrings(value: any[]): string[] {
 	return _.map(value, String);
 }
 
-function validateServiceVolume(serviceVolume: string, volumeNames: string[]) {
-	const colonIndex = serviceVolume.indexOf(':');
-	if (colonIndex === -1) {
-		throw new ValidationError(`Invalid volume: '${serviceVolume}'`);
+function normalizeServiceBuild(
+	serviceBuild: string | BuildConfig,
+	networkNames: string[],
+): BuildConfig {
+	if (typeof serviceBuild === 'string') {
+		serviceBuild = { context: serviceBuild };
+	}
+	if (serviceBuild.args) {
+		serviceBuild.args = normalizeKeyValuePairs(serviceBuild.args);
+	}
+	if (serviceBuild.labels) {
+		serviceBuild.labels = normalizeKeyValuePairs(serviceBuild.labels);
+		validateLabels(serviceBuild.labels);
+	}
+	if (serviceBuild.extra_hosts && !_.isArray(serviceBuild.extra_hosts)) {
+		serviceBuild.extra_hosts = normalizeExtraHostObject(
+			serviceBuild.extra_hosts as any,
+		);
+	}
+	if (serviceBuild.isolation) {
+		throw new ValidationError('service.build.isolation is not allowed');
+	}
+	if (
+		serviceBuild.network &&
+		serviceBuild.network !== 'host' &&
```

Comment:
I just realized that we probably don't want this

Reply:
thinking about this again, I think we don't need to restrict here... downstream components can still decide to apply limits (or ignore altogether). and the cli might want to allow this property anyway...